sdk/python/foundation-models/system/inference/text-to-image/aacs-prepare-uai.ipynb

```diff
@@ -0,0 +1,434 @@
+    "    subscription_id=subscription_id,\n",
+    "    credential=credential,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### 2.2 Create the User Assigned Identity:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "msi_client.user_assigned_identities.create_or_update(\n",
+    "    resource_group_name=resource_group,\n",
+    "    resource_name=uai_name,\n",
+    "    parameters=Identity(location=workspace_location),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### 2.3 Retrieve the identity object\n",
+    "we need to retrieve the identity object so that we can use it to deploy the Azure AI Content Safety enabled text-to-image online endpoint."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "uai_identity = msi_client.user_assigned_identities.get(\n",
+    "    resource_group_name=resource_group,\n",
+    "    resource_name=uai_name,\n",
+    ")\n",
+    "uai_principal_id = uai_identity.principal_id\n",
+    "uai_client_id = uai_identity.client_id\n",
+    "uai_id = uai_identity.id\n",
+    "print(f\"UAI principal id: {uai_principal_id}\")\n",
+    "print(f\"UAI id: {uai_id}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 2.4 Grant appropriate roles to the UAI we created above.\n",
+    "Note: In order to successfully run scripts in current step, you must have owner permission on the Azure AI Content Safety (AACS) resource and the text-to-image endpoint, which we created in the previous steps."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### 2.4.1 Get an AuthorizationManagementClient to list Role Definitions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.mgmt.authorization import AuthorizationManagementClient\n",
+    "from azure.mgmt.authorization.v2022_04_01.models import (\n",
+    "    RoleAssignmentCreateParameters,\n",
+    ")\n",
+    "import uuid\n",
+    "\n",
+    "role_definition_client = AuthorizationManagementClient(\n",
+    "    credential=credential,\n",
+    "    subscription_id=subscription_id,\n",
+    "    api_version=\"2022-04-01\",\n",
+    ")\n",
+    "role_assignment_client = AuthorizationManagementClient(\n",
+    "    credential=credential,\n",
+    "    subscription_id=subscription_id,\n",
+    "    api_version=\"2022-04-01\",\n",
+    ")\n",
+    "\n",
+    "uai_role_check_list = {\n",
+    "    \"Cognitive Services Contributor\": {\n",
+    "        \"step\": \"4.2.2\",\n",
+    "        \"description\": \"assigne the role Cognitive Services User to the UAI on the Azure AI Content Safety resource.\",\n",
```

Comment:
assign Cognitive Services User role to UAI on the Azure AI Content Safety resource